agent/lib/broker_address_settings.js

```diff
@@ -18,65 +18,90 @@
 var log = require("./log.js").logger();
 var kubernetes = require('./kubernetes.js');
 var myutils = require('./utils.js');
+var events = require('events');
+var util = require('util');
 
-function kubernetes_resource_compare(a, b) {
-    return myutils.string_compare(a.metadata.name, b.metadata.name);
+function extract_flatten_plan(def) {
```

Comment:
Is there any real need to 'flatten' the definition? Iirc the reason this is done for addresses is because of legacy configuration format from looong ago.

Reply:
ah, I'll change that.  I saw it was done elsewhere and followed.